sklearn/ensemble/partial_dependence.py

```diff
@@ -306,8 +306,8 @@ def convert_feature(fx):
                 l.append(feature_names[i])
             names.append(l)
     except IndexError:
-        raise ValueError('features[i] must be in [0, n_features) '
-                         'but was %d' % i)
+        raise ValueError('features[i] must be in [0, len(feature_names)={0} ) '
+                         'but was {1}'.format(len(feature_names), i))
```

Comment:
It was a bit of a tricky issue here; perhaps the error message needs further revision. 

I can't show `features[i]` in this case, since `i` is out of range of the array. Showing the value for `len(feature_names` is also a bit confusing, since both seem like they should contain at most `n_features-1` elements.

From what I can tell based on the documentation, `features` is an array of indices for the features whose names correspond to the elements of `feature_names` at `features[i]`. My goal here is to work out a way to show a message such that a user who hasn't fully read the documentation would understand it.